### ISSUE TITLE
git.checkout().setStartPoint(commitCheckout).addPaths(paths).call() seems not to work as expected.

### ISSUE THREAD

**Issue by fermadeiral** [Spirals-Team/repairnator#581]
When trying to solve #490, I got a case when `git.checkout().setStartPoint(commitCheckout).addPaths(paths).call()` seems not to work as we expected.

Example: https://github.com/INRIA/spoon/commit/551f0fb28c49f4b1eeab93f7748c04ef82cfde39 (this commit is a fixer commit).

There are 3 test classes involved:
1) src/test/java/spoon/test/imports/ImportTest.java
2) src/test/java/spoon/test/imports/testclasses/TransportIndicesShardStoresAction.java (NEW)
3) src/test/java/spoon/test/imports/testclasses2/AbstractMapBasedMultimap.java

Consider I have checked out that commit.
Then I have its previous commit called `commitCheckout`, and a variable `paths` containing `src/test/java`.
When I call `git.checkout().setStartPoint(commitCheckout).addPaths(paths).call()`, files 1 and 3 are updated for their previous state, but file 2, the new one, remains checked out.

I see no other solution than discovering the new files and deleting them by force. I also still don't know if the opposite happens (i.e., if a deleted file is not checked out in the commit it existed).

**Comment by surli** [comment]
okay... It might be another behaviour of JGit which is not standard compared to the git one -_-

**Comment by fermadeiral** [comment]
I talked with Thomas about that issue since his knowledge on git is advanced and in fact he showed me that `git checkout <paths>` is not the right command to do the job I want, i.e. the problem is not with JGit. He gave me a sequence of commands that does the job properly:

```
git reset <commit> -- <paths>
git clean -fd
git checkout -- .
```
Thanks @tdurieux 

@surli, a PR is coming.

**Comment by surli** [comment]
Fixed in #592
